### DISCUSSION TITLE
CRL refresh interval configurable.

### DISCUSSION THREAD

**Discussion by sc-sityad** [emqx/emqx#11016] [Q&A] [answered]
Hi @thalesmg 
#10557 
- When I apply above configuration to emqx.conf like `EMQX_CRL_CACHE__REFRESH_INTERVAL=5m `. It pings the CRL url that is `https://localhost:8081/1.0/crl` every five minutes but It did not terminate the client connection. 
- When I use the default settings without editing any config of CRL. It terminates after (15 min  to 30 min) of time. What is the issue I a facing here?

**Comment by thalesmg** [comment, marked as answer]
Hi.
Indeed, even if the CRL is later refreshed, connected clients with now-revoked certificates are not actively disconnected, only new ones.
The termination you mention is not clear if it's after some time of inactivity.  It sounds like it could be the lack of keepalive from the client itself.

**Reply by sc-sityad** [reply]
If I dont use CRL feature and revoked the cert. over 1 hour of inactivity the client is not terminated.
But if I use CRL feature and revoked the cert. over 15 to 30 minutes of inactivity the client is terminated and the client whose certificate is not revoked is still there.  Is it correct?

**Reply by thalesmg** [reply]
Without more detailed info about the setup, client or logs, it's hard to understand the problem. 
If you enable debug logging in EMQX, what are the reasons for disconnection?  Any logs from the client side that could help identify the reason?

**Comment by sc-sityad** [comment]
Hi @thalesmg 
I wanted to ask if a certificate has 1 year validity and It is connected throughout and the certificate is compromised. Then I revoked the certificate. and until the client  disconnects the connection manually it is still connected?  It sounds logically wrong right? I have enabled the CRL check in ssl config. I am using emqx 5.0.26 version.

**Reply by thalesmg** [reply]
OTP (and consequently EMQX) does not iterate over the connections after CRLs are updated.  You can always kick or ban clients using the API, since it's known that the certificate was compromised for them.

**Comment by thalesmg** [comment]
An update: I've not been able to reproduce your issue.  I've run the server for more than 1 hour with default CRL options, and the client was not disconnected, as noted.
